plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/explorer/QStatusBarLoggedInActionGroup.kt

```diff
@@ -64,15 +65,18 @@ class QStatusBarLoggedInActionGroup : DefaultActionGroup() {
             addAll(buildActionListForConnectHelp(actionProvider))
 
             add(Separator.create())
-            add(CodeWhispererShowSettingsAction())
-            (
-                ToolkitConnectionManager.getInstance(it).activeConnectionForFeature(QConnection.getInstance()) as? AwsBearerTokenConnection
-                )?.takeIf { !it.isSono() }
-                ?.let { add(QSwitchProfilesAction()) }
-            ToolkitConnectionManager.getInstance(it).activeConnectionForFeature(CodeWhispererConnection.getInstance())?.let { c ->
-                (c as? AwsBearerTokenConnection)?.let { connection ->
-                    add(SsoLogoutAction(connection))
+            add(actionManager.getAction("codewhisperer.settings"))
+
+            val connection = ToolkitConnectionManager.getInstance(it).activeConnectionForFeature(QConnection.getInstance()) as? AwsBearerTokenConnection
+
+            if (connection != null) {
+                if (!connection.isSono()) {
+                    add(actionManager.getAction("codewhisperer.switchProfiles"))
```

Comment:
Why not keep using `add(QSwitchProfilesAction())`?

Reply:
marginally saves some GC cycles because we dont need to construct/gc